### PR TITLE
Script for generating SECP256K1 vectors

### PR DESCRIPTION
**Pull request by public**
For #1109.

This loads the FIPS 186-3 SigGen.txt for ECDSA and generates an SECP256K1 signature for every hash and message pair.

Currently this verifies the vectors it's generating using our ECDSA API, but that API doesn't actually support SECP256K1 right now so this PR also includes code which enables that.

### CODE REVIEW

File: docs/development/custom-vectors/secp256k1/generate_secp256k1.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+dest_path = os.path.join("asymmetric", "ECDSA", "SECP256K1", "SigGen.txt")
+
+fips_vectors = load_vectors_from_file(
+    source_path,
+    load_fips_ecdsa_signing_vectors
+)
+
+write_file(
+    build_vectors(fips_vectors),
+    open_vector_file(dest_path, "w")
```
Comment (alex) [inline docs/development/custom-vectors/secp256k1/generate_secp256k1.py]: Please close this file explcitly (with statement)

**Comment by alex** [comment]
This definitely needs docs :-)

**Comment by public** [comment]
Are we OK with the general approach here?

**Comment by reaperhulk** [comment]
Do we need to do this? It looks like there are some vectors available already based on #1109.

**Comment by public** [comment]
Sadly none of the existing vectors include the `r` and `s` values we expose in our API, just the internal `k` value.

**Comment by reaperhulk** [comment]
Okay, makes sense. When we've done generation in the past (e.g. #626) we've added a new section to the docs (https://cryptography.io/en/latest/development/test-vectors/#custom-symmetric-vectors) that inlines the generation script as well as adding the verification code and the vectors themselves.

I'd also prefer that the actual code to enable the curve not land in this PR. That does mean that the generation script won't work immediately, but the code to enable the curve should only be added with the tests.

**Comment by public** [comment]
OK as long as we are fine with the idea I'll do some more work on this :)

**Comment by public** [comment]
Updated. Should I remove the SECP256K1 curve definition I added?

**Comment by reaperhulk** [comment]
@public yeah let's remove it until we land support. Looks like you've got a docs failure to fix still.

**Comment by reaperhulk** [comment]
I've verified these scripts generate as expected (using the python-ecdsa package) and they are verified using OpenSSL. Will merge once tests pass.

**Pull request merged**